wordpress-mobile/WordPress-iOS
# Reader: Add new post detail header

Refs #21644 

This adds the new post detail header, now in SwiftUI.

Before | After
-|-
![before_light](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/b5a8d153-f331-4144-8f29-efde9caf6936) | ![after_light](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/0857008d-6657-4b76-a401-6cfb9ac87e61)
![before_dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/3d49ad67-6885-4291-ac08-63cb74f44288) | ![after_dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/1cf8c7fe-8983-4b43-9bcf-a45a7294ef58)

In the code, I've found that there's a thing called an "attribution view" in Reader detail. I haven't figured out what it's used for, or what kind of posts display this view. Will try to figure this out later. 

Furthermore, I've practically recreated the legacy Follow button in SwiftUI, and turns out that in the Reader detail, the Follow button uses a different style for iPhone and iPad. This has been recreated in SwiftUI, although not 100% perfect. I think it should be okay since we will update the button again soon.

I've decided to break my large changes into 3 so it's more manageable to review and test. Some things are not included in this PR:

1. The Follow button is still using the legacy design.
2. The tags list is moved but still uses the legacy design.
3. Some accessibility/VoiceOver polish.
4. Action tab bar.
5. Sans-serif font for the body. Turns out this is quite trivial to do, so we'll be able to ship this in the iteration!

Note: there's also a weird legacy issue where the featured image overlaps into the header area by about 4px. I haven't figured out why this happened, and for now, I've added around 4px additional top padding to the new header. The downside is, this will mean 4px extra padding when the post does not have a featured image.

## To test

- Launch the Jetpack app.
- Turn on the `readerImprovements` feature flag.
- Navigate to Reader.
- Open any Reader post — pick two: one with feature image, and one without.
- 🔎 Verify that the new header design is shown.
- 🔎 Verify that the Follow/Unfollow button works.
- 🔎 Verify that tapping the blog name navigates you to the Site stream.
- Select a post that contains several tags, such that they are collapsed. 
- Tap on the "+" button in the tags list to expand the tags.
- 🔎 Verify that expanding/collapsing the tags works correctly.
- Rotate the device to landscape orientation.
- 🔎 Verify that the layout doesn't break.

Additional variations:
- Test in iPad.
- Test between light/dark schemes.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The feature flag gates all the new codes.

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

7. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
